reference/reference.go

```diff
@@ -35,7 +35,7 @@ const (
 
 var (
 	// ErrReferenceInvalidFormat represents an error while trying to parse a string as a reference.
-	ErrReferenceInvalidFormat = errors.New("invalid reference format")
+	ErrReferenceInvalidFormat = errors.New("invalid reference format: repository name must be lowercase")
```

Comment:
This error is returned when the regex fails for any reason (other than incorrect casing).  Perhaps we could rerun the regex on a lowercased reference and return this verbiage if it passes.